examples/common.cpp

```diff
@@ -95,6 +96,8 @@ std::string process_escapes(const char* input) {
 
 bool gpt_params_parse(int argc, char ** argv, gpt_params & params) {
     bool invalid_param = false;
+    bool escape_prompt = false;
+    const char* prompt = nullptr;
```

Comment:
`process_escapes` takes a C-style string, so `prompt` just holds the unparsed prompt passed in through argv until we know if we want to parse it or assign it to `params.prompt`, which will result in a `std::string`.

It also turns out some people pass entire files in through the prompt, so it also seemed more efficient to do it this way.

Reply:
> It also turns out some people pass entire files in through the prompt, so it also seemed more efficient to do it this way.

but the code does not process files right now.

Reply:
Yeah, sorry if I wasn't clear, people are using the `--prompt` parameter to pass in files rather than using `--file`. I got a toasty message about it, but I'm not one to hold grudges so I figure if people are doing it that way, there's no need to penalize them.